iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -127,12 +228,41 @@ public static void checkHvdcMaxP(Validable validable, double maxP) {
         }
     }
 
+    /**
+     * @deprecated
+     * Use
+     *     {@link #checkRegulatingTerminal(Validable, String, Terminal, boolean, Network)} or
+     *     {@link #checkRegulatingTerminal(Validable, String, Terminal, RegualtionMode, Network)} instead
+     */
+    @Deprecated(since = "4.4.0")
     public static void checkRegulatingTerminal(Validable validable, Terminal regulatingTerminal, Network network) {
         if (regulatingTerminal != null && regulatingTerminal.getVoltageLevel().getNetwork() != network) {
             throw new ValidationException(validable, "regulating terminal is not part of the network");
         }
     }
 
+    /**
+     * For validating regulating terminal.
+     * If control is enabled regulating terminal must be defined and valid (Network associated with it should be ok).
+     */
+    public static void checkRegulatingTerminal(Validable validable, String validableType, Terminal regulatingTerminal,
+        boolean voltageRegulatorOn, Network network) {
+        if (voltageRegulatorOn && !validRegulatingTerminal(regulatingTerminal, network)) {
```

Comment:
Same thing here, it does not make sense to have a regulating terminal in another network even if not regulating

Reply:
Changed. When the control is disabled only valid values and Null (initial value, `regulatingTerminal` is not set) are allowed.